src/search/srpso.c

```
@@ -0,0 +1,239 @@
+
+static void sr_pso_free_swarm(sr_pso_swarm *s)
+{
+  if (!s) return;
+  sr_free_matrix(s->pos);
+  sr_free_matrix(s->vel);
+  sr_free_matrix(s->pbest);
+  sr_free_vector(s->pbest_val);
+  sr_free_vector(s->gbest);
+}
+
+static int sr_pso_alloc_swarm(sr_pso_swarm *s, int swarm, int ndim)
```

Clang (14.0.6)

[warning] parameter name 's' is too short, expected at least 3 characters

(readability-identifier-length)